projects/knora-ui/src/lib/viewer/values/time-value/time-value.component.ts

```diff
@@ -0,0 +1,198 @@
+                       (_.isEqual(initValue.date, this.convertedControlDate) || initValue.date === control.value.date) &&
+                       (initValue.time === control.value.time || control.value.time.match(CustomRegex.TIME_REGEX) == null) &&
+                       (initComment === commentFormControl.value || (initComment === null && commentFormControl.value === '')));
+
+      return invalid ? {valueNotChanged: {value: control.value}} : null;
+    };
+  };
+
+  getInitValue(): DateTime | null {
+    if (this.displayValue !== undefined) {
+      
+      const datePipe = new DatePipe('en-US');
+      this.date = datePipe.transform(this.displayValue.time, "d.M.y");
+      this.time = datePipe.transform(this.displayValue.time, "HH:mm");
+
+      this.initKnoraDate = new KnoraDate("Gregorian",
+                                          "AD",
+                                          Number(datePipe.transform(this.displayValue.time, "y")),
+                                          Number(datePipe.transform(this.displayValue.time, "M")),
+                                          Number(datePipe.transform(this.displayValue.time, "d")));
+
+      return new DateTime(this.initKnoraDate, this.time);
+    } else {
+      return null;
+    }
+  }
+
+  ngOnInit() {
+    // initialize form control elements
+    this.valueFormControl = new FormControl(null);
+
+    this.commentFormControl = new FormControl(null);
+
+    // subscribe to any change on the comment and recheck validity
+    this.valueChangesSubscription = this.commentFormControl.valueChanges.subscribe(
+      data => {
+        this.valueFormControl.updateValueAndValidity();
+      }
+    );
+
+    this.form = this.fb.group({
+      timeValue: this.valueFormControl,
+      comment: this.commentFormControl,
+    });
+
+    this.resetFormControl();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    this.resetFormControl();
+  }
+
+  // unsubscribe when the object is destroyed to prevent memory leaks
+  ngOnDestroy(): void {
+    this.unsubscribeFromValueChanges();
+  }
+
+  getNewValue(): CreateTimeValue | false {
+    if (this.mode !== 'create' || !this.form.valid) {
+      return false;
+    }
+
+    const newTimeValue = new CreateTimeValue();
+
+    // split the time entry in two to separate the hours and the minutes
+    let splitTime = this.valueFormControl.value.time.split(":");
+
+    // when the first value is displayed on page load, it will be an instance of KnoraDate
+    if(this.valueFormControl.value.date instanceof KnoraDate){
+      this.updateDate = new Date(this.valueFormControl.value.date.year,
+                                (this.valueFormControl.value.date.month - 1),
+                                this.valueFormControl.value.date.day,
+                                splitTime[0],
+                                splitTime[1]
+      );
+    } else { // when the user submits a new value, it will be an instance of GregorianCalendarDate
```

Comment:
Could you make this consistent with the member `date` of `DateTime`? I.e. `TimeInputComponent` should receive and return a `KnoraDate` (conversion of a GregorianCalendarDate to a `KnoraDate` has to be handled by the value getter). So `TimeInputComponent` would handle the complexity of conversion.

Reply:
Fixed in 046010e. The child component now does all the work.